### PR TITLE
ci: fix pr-review handling of review comments

### PR DESCRIPTION
**Pull request by wexxlee**
GH apparently fires a `pull_request_review` event on comments/replies to existing reviews and treats each as a new review (via the `reviews` endpoint).  This fixes an edge case where the `needs-review` tag is incorrectly removed when there has been an unreviewed commit submitted, but a comment/reply to an earlier review is posted after the commit.

(I am rapidly developing an "appreciation" 🙄 for the idiosyncrasies of GH workflow/event logic.)

### CODE REVIEW

**Review by JLGarber — APPROVED** [review]

**Comment by JLGarber** [comment]
Goodness me that's a lot of fine detail to work through. Thanks for hunting all this stuff!

**Comment by wexxlee** [comment]
I'm going to hold off on merging this for now.  Unrelated to this PR, but related to the same workflow job - the workflow seems to occasionally fail right now when review comments are posted, but only intermittently.  (See, e.g., [here](https://github.com/OverlayPlugin/cactbot/actions/runs/8352969931/job/22863922979).)  The error message claims that the .cjs script is exceeding the secondary rate limit for the REST API, but we are *nowhere remotely within the same universe* of the applicable rate limit -- and the x-http headers in the error message show we haven't exceeded the rate limit).  I can also hammer the API repeatedly by running the script back-to-back from the CLI without getting the same error, so it seems to be something happening as a result of running the script from within a workflow job.

I've got a ticket open with GH right now to try to nail down why the rate limit error is showing up, so I'm going to try to run that to ground and see if we need a code fix before merging this.

**Comment by wexxlee** [comment]
> I've got a ticket open with GH right now to try to nail down why the rate limit error is showing up, so I'm going to try to run that to ground and see if we need a code fix before merging this.

We've narrowed the issue to an aspect of the secondary rate limit that involves queries to the REST API which are compute-expensive.  (Aside: It's not at all clear to me (or GH support) why a single query to the search endpoint with several well-defined filters would be particularly expensive, or why the first instance of the query would trigger the rate limit, but 🤷).  Apparently, there are also different thresholds for `secrets.GITHUB_TOKEN` as opposed to a run-of-the-mill PAT -- which is why you can run the .cjs script locally with a PAT and absolutely hammer the endpoint without tripping the same anti-abuse threshold -- a fact that is mystifying to both me and the GH folks I'm talking to.

It's been escalated to GH eng., but I'm a little skeptical right now about getting a clear answer/fix.  So we may have to revert back to querying the `pulls` endpoint and iterating through all of the pulls to find ones with the correct label.  (Which, I would think, would be way more expensive for GH for both API load and for processing time on the runner, but what do I know...)

**Comment by wexxlee** [comment]
From GH:

> To be completely transparent, this involves multiple engineering teams and further investigation might take a while.
>
> There isn't other way to fetch PRs directly for a specific label. My recommendation is the same as what you noted earlier -- **we'd have to hit the pulls endpoint, grab all pulls, and then iterate over them, which seems much more resource-costly.**

I'll go ahead and switch back to the version of the code that iterated over the `pulls` endpoint (I think it was one of the earlier commits on the original PR before I switched it to using the `search` endpoint).

@valarnin, just an FYI since I think you'd originally asked to limit the API request by label.  Lmk if you have concerns.

**Comment by wexxlee** [comment]
Aha! I found another way.  Based on this gem buried in the [API spec](https://octokit.github.io/rest.js/v20) about the `issues` endpoint:

> Note: GitHub's REST API considers every pull request an issue, but not every issue is a pull request. For this reason, "Issues" endpoints may return both issues and pull requests in the response. You can identify pull requests by the pull_request key.

(in case you actually go read that, ignore the next following bit about issue ids != pr ids; we don't use `id`, just `number`, which is consistent across both endpoints)

I updated it to use the `issues` endpoint.  Now grabs just the PRs by label.  Tested and working.

(Of course, whether this will still occasionally fail because of secondary rate limit is another question, but fingers crossed this is fixed now.)

**Comment by JLGarber** [comment]
Thanks for handling the escalation with Github. It's maybe a little frustrating that they haven't cleaned up that bit of technical debt by now, but at least there's a workaround.

**Pull request merged**
